src/python_framework_v02/troute/nhd_io.py

```diff
@@ -233,6 +238,98 @@ def get_ql_from_wrf_hydro(qlat_files, index_col="station_id", value_col="q_later
 
     return ql
 
+def read_netcdfs(files, dim, transform_func=None):
+    def process_one_path(path):
+        with xr.open_dataset(path) as ds:
+            if transform_func is not None:
+                ds = transform_func(ds)
+            ds.load()
+            return ds
+
+    paths = sorted(glob.glob(files))
+    datasets = [process_one_path(p) for p in paths]
+    combined = xr.concat(datasets, dim)
+    return combined
+
+def add_time_dimension(ds):
```

Comment:
We'll want to use the new function, "preprocess_time_station_index"

Reply:
I'm not finding this in the code anywhere. I pulled in any updates.

Reply:
It's in the new notebook.